hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/AbstractHawkbitUI.java

```diff
@@ -164,7 +168,7 @@ public void afterViewChange(final ViewChangeEvent event) {
         navigator.addView(EMPTY_VIEW, new Navigator.EmptyView());
         // set locale is required for I18N class also, to get the locale from
         // cookie
-        final String locale = getLocaleId(SPUIDefinitions.getAvailableLocales());
+        final String locale = getLocaleId(new HashSet<>(uiProperties.getLocalization().getAvailableLocals()));
```

Comment:
If you'll change signature of `getLocaleId(Set<String)` to `getLocaleId(List<String)` there is no need for conversion  to HashSet